### ISSUE TITLE
Get package overrides from txt file in targeting pack instead of msbuild item

### ISSUE THREAD

**Issue by nguerrera** [dotnet/sdk#2937]
https://github.com/aspnet/AspNetCore/pull/7555

The file format is very similar to the item.

Before: build\netcoreapp3.0\Microsoft.AspNetCore.App.props:
 ```xml
<ItemGroup>
   <PackageConflictOverrides Include="Microsoft.AspNetCore.App">
       <OverriddenPackages>
         Microsoft.AspNetCore.Antiforgery|3.0.0-preview3-19111-04;
         Microsoft.AspNetCore.Authentication.Abstractions|3.0.0-preview3-19111-04;
         Microsoft.AspNetCore.Authentication.Cookies|3.0.0-preview3-19111-04;
         // ...
 
```              
 

After: build\netcoreapp3.0\Microsoft.AspNetCore.App.PackageOverrides.txt:

```
Microsoft.AspNetCore.Antiforgery|3.0.0-preview3-19111-04
Microsoft.AspNetCore.Authentication.Abstractions|3.0.0-preview3-19111-04
Microsoft.AspNetCore.Authentication.Cookies|3.0.0-preview3-19111-04
// ...
```

The identity of the item follows from the package/file name, the metadata is the body of the txt file, where newlines delimit instead of `;` with whitespace ignored.

**Comment by nguerrera** [comment]
@dsplaisted @natemcmaster

**Comment by nguerrera** [comment]
cc @dagood @ericstj 

There is also PackageConflictPreferredPackages that needs to be replaced. I think it can be converted to PackageConflictOverrides. It looks like the former predates the latter and is used by the base shared framework TP. However, since it doesn't specify versions, we have to crack files. The newer way gives a perf optimization.

I think we can do the following:

1. Microsoft.NETCore.App 3.0 and NETStandard.Library 2.1 provide a *.PackageOverrides like the linked one for ASP.NET
2. We implement readng these txt files
3. We delete this:

https://github.com/dotnet/sdk/blob/f0a1246be1ebe11a88097ac3dd5736f0ad15b855/src/Tasks/Microsoft.NET.Build.Tasks/ResolveTargetingPackAssets.cs#L66-L68

4. We disable this for anything but NETCoreApp < 3.0 and NETStandard < 2.1:
https://github.com/dotnet/sdk/blob/2eb6c546931b5bcb92cd3128b93932a980553ea1/src/Tasks/Common/targets/Microsoft.NET.DefaultPackageConflictOverrides.targets

**Comment by dsplaisted** [comment]
We're now thinking of `data/PackageOverrides.txt` for the path to this file in the package.

**Comment by dsplaisted** [comment]
Copying from https://github.com/aspnet/AspNetCore/issues/8836#issuecomment-481006077:

We suggest that eventually we should use file names in these packages that don't embed the package / shared framework name, IE:

- `data\PackageOverrides.txt`
- `data\PlatformManifest.txt`

**Comment by livarcocc** [comment]
@dsplaisted said this is likely a dup. Please go through them and close duplicates.

**Comment by dsplaisted** [comment]
Closing this in favor of dotnet/cli#10581
